src/utils/useWindowResizeDebounce.ts

```diff
@@ -0,0 +1,19 @@
+import { useEffect, useRef } from 'react';
+
+export const useWindowResizeDebounce = (callback: any, delay: number): void => {
+  const timeout = useRef<number | null>(null);
```

Comment:
I don't see a need to use `null` here.

```suggestion
  const timeout = useRef<number>();
```

Reply:
You'll have to initialize the state with a valid number in that case though, otherwise the compiler will complain

Reply:
In that case just use undefined

timeout = useRef<number|undefined>()